diff --git a/src/desktop_file.rs b/src/desktop_file.rs
index e071bb5..57991e4 100644
--- a/src/desktop_file.rs
+++ b/src/desktop_file.rs
@@ -25,6 +25,7 @@ fn find_application_dirs() -> Vec<PathBuf> {
     let mut dirs = vec![
         PathBuf::from("/usr/share/applications"), // system installed apps
         PathBuf::from("/var/lib/flatpak/exports/share/applications"), // flatpak apps
+        PathBuf::from("/run/current-system/sw/share/applications"), // nixos apps
     ];
 
     let user_dir = dirs::data_local_dir(); // user installed apps
