From 99e3789e5cb9df32a0dbd201a656303a92c662f7 Mon Sep 17 00:00:00 2001
From: Christoph Asche <christoph@asche.co>
Date: Wed, 10 May 2023 23:28:09 +0200
Subject: [PATCH] Add MagentaTV One 2

---
 androidtv/basetv/basetv.py | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/androidtv/basetv/basetv.py b/androidtv/basetv/basetv.py
index 33de34c..e4e25d0 100644
--- a/androidtv/basetv/basetv.py
+++ b/androidtv/basetv/basetv.py
@@ -153,6 +153,15 @@ class BaseTV(object):  # pylint: disable=too-few-public-methods
         ):
             return constants.CMD_CURRENT_APP_GOOGLE_TV
 
+
+        # Is this a Magenta TV One?
+        if (
+            self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV
+            and "SkyworthDigital" in self.device_properties.get("manufacturer", "")
+            and "MagentaTV ONE" in self.device_properties.get("model", "")
+        ):
+            return constants.CMD_CURRENT_APP_GOOGLE_TV
+
         # Is this an Android 11 device?
         if self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV and self.device_properties.get("sw_version", "") == "11":
             return constants.CMD_CURRENT_APP11
@@ -171,7 +180,7 @@ class BaseTV(object):  # pylint: disable=too-few-public-methods
         if constants.CUSTOM_CURRENT_APP_MEDIA_SESSION_STATE in self._custom_commands:
             return self._custom_commands[constants.CUSTOM_CURRENT_APP_MEDIA_SESSION_STATE]
 
-        # Is this a Google Chromecast Android TV?
+        # Is this a Magenta TV One?
         if (
             self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV
             and "SkyworthDigital" in self.device_properties.get("manufacturer", "")
@@ -179,7 +188,7 @@ class BaseTV(object):  # pylint: disable=too-few-public-methods
         ):
             return constants.CMD_CURRENT_APP_MEDIA_SESSION_STATE_GOOGLE_TV
 
-        # Is this a Magenta TV One?
+        # Is this a Google Chromecast Android TV?
         if (
             self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV
             and "Google" in self.device_properties.get("manufacturer", "")
@@ -236,6 +245,14 @@ class BaseTV(object):  # pylint: disable=too-few-public-methods
         ):
             return constants.CMD_LAUNCH_APP_GOOGLE_TV.format(app)
 
+        # Is this a Magenta TV One?
+        if (
+            self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV
+            and "SkyworthDigital" in self.device_properties.get("manufacturer", "")
+            and "MagentaTV ONE" in self.device_properties.get("model", "")
+        ):
+            return constants.CMD_LAUNCH_APP_GOOGLE_TV.format(app)
+
         if self.DEVICE_ENUM == constants.DeviceEnum.FIRETV:
             return constants.CMD_LAUNCH_APP_FIRETV.format(app)
 
-- 
2.40.1

