From afb4f0e2022a90239e73b6fe386b9255266e7cbe Mon Sep 17 00:00:00 2001
From: Christoph Asche <christoph@asche.co>
Date: Wed, 10 May 2023 22:53:43 +0200
Subject: [PATCH] Add MagentaTV One

---
 androidtv/basetv/basetv.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/androidtv/basetv/basetv.py b/androidtv/basetv/basetv.py
index 0f89589..33de34c 100644
--- a/androidtv/basetv/basetv.py
+++ b/androidtv/basetv/basetv.py
@@ -172,6 +172,14 @@ class BaseTV(object):  # pylint: disable=too-few-public-methods
             return self._custom_commands[constants.CUSTOM_CURRENT_APP_MEDIA_SESSION_STATE]
 
         # Is this a Google Chromecast Android TV?
+        if (
+            self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV
+            and "SkyworthDigital" in self.device_properties.get("manufacturer", "")
+            and "MagentaTV ONE" in self.device_properties.get("model", "")
+        ):
+            return constants.CMD_CURRENT_APP_MEDIA_SESSION_STATE_GOOGLE_TV
+
+        # Is this a Magenta TV One?
         if (
             self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV
             and "Google" in self.device_properties.get("manufacturer", "")
-- 
2.40.1

